api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,206 @@
+	// in parallel to openshift-router.
+	//
+	// +kubebuilder:default:={}
+	// +kubebuilder:validation:Optional
+	// +optional
+	AdditionalResourceTags map[string]string `json:"additionalResourceTags,omitempty"`
+
+	// IngressClass specifies the Ingress class which the controller will reconcile.
+	// This Ingress class will be created unless it already exists.
+	// The value will default to "alb".
+	//
+	// +kubebuilder:default:=alb
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
+
+	// Config specifies further customization options for the controller's deployment spec.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Config *AWSLoadBalancerDeploymentConfig `json:"config,omitempty"`
+
+	// AWSAddon describes the AWS services that can be integrated with
+	// the AWS Load Balancer.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	EnabledAddons []AWSAddon `json:"enabledAddons,omitempty"` // indicates which AWS addons should be disabled.
+
+	// Credentials is a reference to a secret containing
+	// the AWS credentials to be used by the controller.
+	// The secret is required to be in the operator namespace.
+	// If this field is empty - the credentials will be
+	// requested using the Cloud Credentials API,
+	// see https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.html.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Credentials *SecretReference `json:"credentials,omitempty"`
+}
+
+type AWSLoadBalancerDeploymentConfig struct {
+
+	// +kubebuilder:default:=2
+	// +kubebuilder:validation:Optional
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+}
+
+// SecretReference contains the information to let you locate the desired secret.
+// Secret is required to be in the operator namespace.
+type SecretReference struct {
+	// Name is the name of the secret.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// AWSLoadBalancerControllerStatus defines the observed state of AWSLoadBalancerController.
+type AWSLoadBalancerControllerStatus struct {
+
+	// Conditions is a list of operator-specific conditions
+	// and their status.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the most recent generation observed.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Subnets contains details of the subnets of the cluster
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Subnets *AWSLoadBalancerControllerStatusSubnets `json:"subnets,omitempty"`
+
+	// IngressClass is the current default Ingress class.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
```

Comment:
Is it guaranteed you will always default to `alb`? Or, can the default value change over time? Preferably, the default 
value should be set in here so it is part of the API. Instead of relying on the operator implementation to perform the defaulting.

Reply:
> EDIT: I just noticed you already define the default value in the spec part.

Yes, the default to `alb` is part of the API.